src/core/simulation_results.jl

```diff
@@ -1,517 +1,420 @@
+    # Accepted Key Words
+    - `initial_time::Dates.DateTime` : initial of the requested results
+    - `count::Int`: Number of results
+"""
+function get_dual_values(res::SimulationResults, name::Symbol; kwargs...)
+    return get_duals_values(res, [name]; kwargs...)[name]
 end
 
-function get_reference(
-    sim_results::SimulationResultsReference,
-    stage::String,
-    step::Int,
-    variable::Symbol,
-)
-    file_paths = sim_results.ref["stage-$stage"][variable]
-    return filter(file_paths -> file_paths.Step == "step-$step", file_paths)[:, :File_Path]
+"""
+    Returns the values for the requested parameter name. It keeps requests when performing multiple retrievals. Accepts a parameter name to return the result.
+    # Accepted Key Words
+    - `initial_time::Dates.DateTime` : initial of the requested results
+    - `count::Int`: Number of results
+"""
+function get_parameter_values(res::SimulationResults, name::Symbol; kwargs...)
+    return get_parameters_values(res, [name]; kwargs...)[name]
 end
 
-get_psi_container(sim::Simulation, stage::Any) = sim.stages[stage].internal.psi_container
+function _store_result!(result_dict, results)
+    for (k, v) in results
+        result_dict[k] = v
+    end
+    return
+end
 
 """
-    write_results(results::SimulationResults)
-
-Exports Simulations Results to the path where they come from in the results folder
-
-# Arguments
-- `results::SimulationResults`: results from the simulation
-- `save_path::String`: folder path where the files will be written
-- `results_folder`: name of the folder where the results will be written
-
-# Accepted Key Words
-- `file_type = CSV`: only CSV and featherfile are accepted
+    Loads the simulation results into memory for repeated reads. Running this function twice
+    overwrites the previously loaded results
+
+    # Required Key Words
+    - `initial_time::Dates.DateTime` : initial of the requested results
+    - `count::Int`: Number of results
+    # Accepted Key Words
+    - `variables::Vector{Symbol}`: Variables names to load into results
+    - `duals::Vector{Symbol}`: Dual names to load into results
+    - `parameters::Vector{Symbol}`: Parameter names to load into results
 """
-function IS.write_results(res::SimulationResults; kwargs...)
-    folder_path = res.results_folder
-    if !isdir(folder_path)
-        throw(IS.ConflictingInputsError("Specified path is not valid. Set up results folder."))
+function load_simulation_results!(
+    res::SimulationResults;
+    initial_time::Dates.DateTime,
+    count::Int,
+    variables::Vector{Symbol} = Symbol[],
+    duals::Vector{Symbol} = Symbol[],
+    parameters::Vector{Symbol} = Symbol[],
+)
+    res.results_timestamps = _process_timestamps(res, initial_time, count)
+    _store_result!(
+        res.variable_values,
+        _get_variables_values(res, variables, res.results_timestamps),
+    )
```

Comment:
Would this work?
```suggestion
   merge!(
        res.variable_values,
        _get_variables_values(res, variables, res.results_timestamps),
    )
```

Reply:
merge will iterate over all the keys in both. _store_result! just searches for the keys in the dict with the values we want to store. I can try it but it seemed better to iterate over the smallest set.

Reply:
I don't think that it correct. I think that it will do exactly what you are doing.
```
merge!(d::AbstractDict, others::AbstractDict...)

Update collection with pairs from the other collections.
```